library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java

```diff
@@ -101,22 +158,60 @@ public WeekViewEvent(long id, String name, String location, Calendar startTime,
      * @param location The location of the event.
      * @param startTime The time when the event starts.
      * @param endTime The time when the event ends.
+     * @param allDay Is the event an all day event
      */
-    public WeekViewEvent(long id, String name, String location, Calendar startTime, Calendar endTime) {
+    @Deprecated
+    public WeekViewEvent(long id, String name, String location, Calendar startTime, Calendar endTime, boolean allDay) {
+        this(Integer.toString((int) id), name, location, startTime, endTime, allDay, null);
```

Comment:
This one is ok, but doesn't actually need to convert it to a string, the conversion would happen later the other constructor.